tests/run-build.sh

```diff
@@ -38,8 +38,10 @@ if [ "${SEMAPHORE-}" == true ] ; then
         fi
 fi
 
-if [ ${SRC_CHANGES} -eq 0 ] && [ ${DOC_CHANGES} -eq 0 ]; then
-    echo No changes detected.
+HEAD=`git rev-parse HEAD`
+MASTER=`git rev-parse origin/master`
+if [[ ${HEAD} != ${MASTER} ]] && [[ ${SRC_CHANGES} -eq 0 && ${DOC_CHANGES} -eq 0 ]]; then
```

Comment:
Looks like it. I guess that instead of modifying this `if` we could add a new one before, like:
```bash
if [[ ${HEAD} = ${MASTER} ]]; then
    SRC_CHANGES=1
    DOC_CHANGES=1
fi
```